CDCgov/prime-reportstream
# Unwanted conditions to notes schema

This PR adds a schema for converting unwanted conditions to notes to be used with MARS OTC data

Test Steps:
1. Add the following rules to a receiver transform
```
  - name: observation-result-with-aoe
    resource:
      - "%resource.result.resolve().where(code.coding.extension('https://reportstream.cdc.gov/fhir/StructureDefinition/condition-code').value.where(code in ('840539006))) | %resource.basedOn.resolve().supportingInfo.resolve()"

  - name: order-note
    resource: "%resource.basedOn.resolve().note | %resource.result.resolve().where(code.coding.extension('https://reportstream.cdc.gov/fhir/StructureDefinition/condition-code').value.where(code in ('840539006')).exists().not())"
    schema: classpath:/metadata/hl7_mapping/receivers/Common/observation-to-nte/unwanted-conditions-to-notes.yml
```
Here we are indicating that all results that do not have a code of '840539006' (COVID) should go through the new `unwanted-conditions-to-notes` schema

2. Run a multiplex test and verify that all other conditions except for COVID are shown as NTE segments on the resulting message.

## Changes
- added `unwanted-conditions-to-notes` schema
- added integration test

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue
